noncer.go

```diff
@@ -0,0 +1,68 @@
+package oauth1
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+)
+
+// Noncer provides random nonce strings.
+type Noncer interface {
+	Nonce() string
+}
+
+// NoncerFunc is an adapter to allow the use of
+// ordinary functions as Noncers. If f is a function
+// with the appropriate signature, NoncerFunc(f) is a
+// Noncer that calls f.
+type NoncerFunc func() string
+
+// Nonce calls f().
+func (f NoncerFunc) Nonce() string {
+	return f()
+}
+
+var (
+	// DefaultNoncer is the default Noncer. It reads 32
+	// bytes from crypto/rand and returns those bytes as a
+	// base64 encoded string.
+	DefaultNoncer Noncer = Base64Noncer{
+		Length: 32,
+	}
+)
+
+// Base64Noncer reads Length bytes from crypto/rand and
+// returns those bytes as a base64 encoded string. If
+// Length is 0, 32 bytes are read.
+type Base64Noncer struct {
+	Length int
```

Comment:
Is there a true need for altering the length? Every exported type and field is a library commitment.

Reply:
Or if someone sets it below 32 there may be some concern about lack of uniqueness among nonces. The spec doesn't provide a hard requirement.

Reply:
That sounds good to me. I'm not invested either way.